src/globus_cli/termio/context.py

```diff
@@ -94,10 +95,12 @@ def env_interactive() -> bool | None:
 
 
 def term_is_interactive() -> bool:
-    try:
+    with suppress(ValueError):
         env = env_interactive()
         if env is not None:
             return env
-    except ValueError:
-        pass
+
+    if sys.__stdin__.isatty():
```

Comment:
This isn't strictly necessary, but I think using the non-dunder name here might better match any expectations we would have when mocking or otherwise testing this.

```suggestion
    if sys.stdin.isatty():
```

Both `__stdin__` and `stdin` are documented, and `sys.stdin is sys.__stdin__` is true unless `stdin` has been rewritten, so it hardly makes a difference...

But I would be curious to see if this fixes the pytest-under-windows issue. They may be rewriting `stdin` but not `__stdin__` in that scenario.

Worst case, we can go to the failing test and add a call to monkeypatch to set `isatty` to `lambda: True` on the necessary object.

Reply:
I'm seeing some Windows tests passing, so it looks like your hunch was right! Great call 🚀